cpp/src/arrow/flight/sql/odbc/tests/connection_test.cc

```diff
@@ -81,6 +86,54 @@ TEST(SQLAllocConnect, TestSQLAllocHandleConnect) {
   EXPECT_TRUE(return_alloc_connect == SQL_SUCCESS);
 }
 
+TEST(SQLDriverConnect, TestSQLDriverConnect) {
+  // ODBC Environment
+  SQLHENV env;
+  SQLHDBC conn;
+
+  // Allocate an environment handle
+  SQLRETURN ret = SQLAllocEnv(&env);
+
+  EXPECT_TRUE(ret == SQL_SUCCESS);
+
+  ret = SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION, (void*)SQL_OV_ODBC3, 0);
+
+  EXPECT_TRUE(ret == SQL_SUCCESS);
+
+  // Allocate a connection using alloc handle
+  ret = SQLAllocHandle(SQL_HANDLE_DBC, env, &conn);
+
+  EXPECT_TRUE(ret == SQL_SUCCESS);
+
+  // Connect string
+  ASSERT_OK_AND_ASSIGN(std::string connect_str,
+                       arrow::internal::GetEnvVar("ARROW_FLIGHT_SQL_ODBC_CONN"));
+  std::vector<SQLWCHAR> connect_str0(connect_str.begin(), connect_str.end());
+
+  SQLWCHAR outstr[ODBC_BUFFER_SIZE];
+  SQLSMALLINT outstrlen;
+
+  // Connecting to ODBC server.
+  ret = SQLDriverConnect(
+      conn, NULL, &connect_str0[0], static_cast<SQLSMALLINT>(connect_str0.size()), outstr,
+      ODBC_BUFFER_SIZE, &outstrlen, SQL_DRIVER_COMPLETE);
+
+  if (ret != SQL_SUCCESS) {
+    std::cerr << GetOdbcErrorMessage(SQL_HANDLE_DBC, conn) << std::endl;
+  }
+
+  EXPECT_TRUE(ret == SQL_SUCCESS);
+
+  // Disconnect from ODBC
+  ret = SQLDisconnect(conn);
+
+  if (ret != SQL_SUCCESS) {
+    std::cerr << GetOdbcErrorMessage(SQL_HANDLE_DBC, conn) << std::endl;
+  }
+
+  EXPECT_TRUE(ret == SQL_SUCCESS);
```

Comment:
Need to free the connection and environment handles after SQLDisconnect.

Reply:
Yup, added